compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -438,6 +444,56 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 self.write_scalar(Scalar::from_target_usize(align.bytes(), self), dest)?;
             }
 
+            sym::minnumf16 | sym::maxnumf16 | sym::copysignf16 => {
+                let op = match intrinsic_name {
+                    sym::minnumf16 => FloatBinIntrinsic::Min,
+                    sym::maxnumf16 => FloatBinIntrinsic::Max,
+                    sym::copysignf16 => FloatBinIntrinsic::Copysign,
+                    _ => bug!(),
+                };
+                self.float_bin_op_intrinsic::<rustc_apfloat::ieee::Half>(op, args, dest)?;
```

Comment:
Nit: If you import `rustc_apfloat::ieee::{Half, Single, Double, Quad}` then is it's probably just as compact to have flat match branches, rather than nesting a `match` and having the symbol names repeated twice.

```rust
            sym::minnumf16 => self.float_bin_op_intrinsic::<Half>(FloatBinIntrinsic::Min, args, dest)?,
            sym::minnumf32 => self.float_bin_op_intrinsic::<Single>(FloatBinIntrinsic::Min, args, dest)?,
            sym::minnumf64 => self.float_bin_op_intrinsic::<Double>(FloatBinIntrinsic::Min, args, dest)?,
            sym::minnumf128=> self.float_bin_op_intrinsic::<Quad>(FloatBinIntrinsic::Min, args, dest)?,
            // ...
```

Reply:
`fmt` won't keep them single line :(

```rust
sym::minnumf16 => {
    self.float_bin_op_intrinsic::<Half>(FloatBinIntrinsic::Min, args, dest)?
}
sym::minnumf32 => {
    self.float_bin_op_intrinsic::<Single>(FloatBinIntrinsic::Min, args, dest)?
}
sym::minnumf64 => {
    self.float_bin_op_intrinsic::<Double>(FloatBinIntrinsic::Min, args, dest)?
}
sym::minnumf128 => {
    self.float_bin_op_intrinsic::<Quad>(FloatBinIntrinsic::Min, args, dest)?
}
```

Another option would be to use a dedicated method for each operation, which would avoid the lengthy `FloatBinIntrinsic::...`.

```rust
sym::minnumf16 => self.float_min_intrinsic::<Half>(args, dest)?,
sym::minnumf32 => self.float_min_intrinsic::<Single>(args, dest)?,
sym::minnumf64 => self.float_min_intrinsic::<Double>(args, dest)?,
sym::minnumf128 => self.float_min_intrinsic::<Quad>(args, dest)?,
```

Reply:
I think that looks cleaner, nice.